client/src/screens/search-groups.screen.js

```diff
@@ -0,0 +1,265 @@
+import PropTypes from 'prop-types';
+import React, { Component } from 'react';
+import {
+  FlatList,
+  ActivityIndicator,
+  Button,
+  Image,
+  StyleSheet,
+  Text,
+  TouchableHighlight,
+  View,
+} from 'react-native';
+import { graphql, compose } from 'react-apollo';
+import moment from 'moment';
+import Icon from 'react-native-vector-icons/FontAwesome';
+import { connect } from 'react-redux';
+
+import { extendAppStyleSheet } from './style-sheet';
+import ALL_GROUPS_QUERY from '../graphql/all-groups.query';
+import JOIN_GROUP_QUERY from '../graphql/join-group.mutation';
+
+
+const styles = extendAppStyleSheet({
+  groupContainer: {
+    flex: 1,
+    flexDirection: 'row',
+    alignItems: 'center',
+    backgroundColor: 'white',
+    borderBottomColor: '#eee',
+    borderBottomWidth: 1,
+    paddingHorizontal: 12,
+    paddingVertical: 8,
+  },
+  groupName: {
+    fontWeight: 'bold',
+    flex: 0.7,
+  },
+  groupTextContainer: {
+    flex: 1,
+    flexDirection: 'column',
+    paddingLeft: 6,
+  },
+  groupText: {
+    color: '#8c8c8c',
+  },
+  groupImage: {
+    width: 54,
+    height: 54,
+    borderRadius: 27,
+  },
+  groupTitleContainer: {
+    flexDirection: 'row',
+  },
+  groupLastUpdated: {
+    flex: 0.3,
+    color: '#8c8c8c',
+    fontSize: 11,
+    textAlign: 'right',
+  },
+  groupUsername: {
+    paddingVertical: 4,
+  },
+  header: {
+    flexDirection: 'row',
+    justifyContent: 'space-between',
+  },
+});
+
+// format createdAt with moment
+const formatCreatedAt = createdAt => moment(createdAt).calendar(null, {
+  sameDay: '[Today]',
+  nextDay: '[Tomorrow]',
+  nextWeek: 'dddd',
+  lastDay: '[Yesterday]',
+  lastWeek: 'dddd',
+  sameElse: 'DD/MM/YYYY',
+});
+
+class Group extends Component {
+  constructor(props) {
+    super(props);
+    this.myGroups = this.props.myGroups;
+    this.joinGroupQuery = this.props.joinGroupQuery.bind(this);
+    this.joinGroup = this.props.joinGroup.bind(this, this.props.group);
+
+  }
+
+  render() {
+    const { id, name } = this.props.group;
+    console.log(this.myGroups)
+    const tags = (this.props.group.tags.map((elem) => {return '#'+elem.name})).join(',')
+    //color the already subscribed groups green
+    let iHaveDisAlready = false
+    this.myGroups.some(function(a) {
```

Comment:
couldn't you do something like this? you might need to use `_.any`
```
const memberAlready = this.myGroups.any(g => a.id == id);
```

Reply:
Possibly, but is there any reason to given what ive written is correct?

(short of making it ES6 with a fat arrow)

Reply:
Mine is about 5 lines shorter and less complicated.